\documentclass[11pt,a4paper]{article}
\usepackage{color}
\newcommand{\red}[1]{\textcolor{red}{#1}}
\newcommand{\blue}[1]{\textcolor{blue}{#1}}
\newcommand{\green}[1]{\textcolor{green}{#1}}

\usepackage[utf8]{inputenc}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{graphicx}
\usepackage[font=small,skip=0pt]{caption}
\usepackage[left=2.5cm,right=2.5cm,top=2.5cm,bottom=2.5cm]{geometry}
\author{Silvano Garnerone}
\title{BLISS analysis}
\begin{document}
\maketitle
\section{RM79}
U2OS cells treated with Etop, trypsinzed and spotted on PLL coverslips.
The barcoding is explained in the following table, where the third column contains the number of unique molecules of the dataset:

\begin{tabular}{|c|c|c|}
\hline 
DMSO & GTCGTATC & 6340\\ 
\hline 
ETO 10uM & TGATGATC & 12522\\ 
\hline 
ETO 150uM & GTCGTTCC & 54467 \\ 
\hline 
\end{tabular} 

Fig.\ref{fig:rm79_umi_stat} shows the frequency of UMI counts. Most of the sites 
have few UMI localized on them, meaning that few cells are seen to be cut in the same location. The drug has the effect of increasing the number of sites which 
are shared by more cells.
\begin{figure}[hbtp]
\centering
\begin{tabular}{@{}cc@{}}
  \includegraphics[width=.5\textwidth]{rm79_umi_scaling.pdf} &
  \includegraphics[width=.5\textwidth]{rm79_umi_scaling_thr.pdf}
\end{tabular}
\caption{Log-log plot of the UMI scaling. The plot on the left shows the distribution of UMI, while the plot on the right shows the number of occurrences with at least a given number of UMI (each location with at most 2000 UMI, to discard possible outliers). The original dataset of read-counts has not been binned.}
\label{fig:rm79_umi_stat}
\end{figure}

\section{RM53}
Using treated and untreated U2OS cells with 10uM Etoposide drug to induce DSBs, in order to measure drug-induced DSBs by BLISS. This is the barcoding and the counting of unique molecules:

\begin{tabular}{|c|c|c|}
\hline 
DMSO & CATCACGC & 37833\\ 
\hline 
ETO 10uM & TGATGCGC & 61027\\ 
\hline 
\end{tabular} 

The library was sequenced in three different runs and we report the merged output of the analysis, obtained after pooling the fastq files of the paired-end runs, and the bed file (before UMI-filtering) obtained from these latter with the bed file (before UMI-filtering) obtained from the single-end run. 
Fig.\ref{fig:rm53_umi_distro} shows the distribution of UMI counts and the difference between treated and non-treated.

\begin{figure}[hbtp]
\centering
\begin{tabular}{@{}cc@{}}
  \includegraphics[width=.5\textwidth]{rm53_umi_count.pdf} &
  \includegraphics[width=.5\textwidth]{rm53_umi_scaling_threshold.pdf}
\end{tabular}
\caption{Log-log plot of the UMI scaling. The plot on the left shows the distribution of UMI, while the plot on the right shows the number of occurrences with at least a given number of UMI (each location with at most 1000 UMI, to discard possible outliers). The original dataset of read-counts has not been binned.}
\label{fig:rm53_umi_distro}
\end{figure}

Since RM53 and RM79 can be considered biological replicas, when one compares 
Fig.\ref{fig:rm53_umi_distro} with Fig.\ref{fig:rm79_umi_stat} it is important 
to consider that: 1) the starting amount of material could be different; 2) the 
two sequencing experiments have different depth (as suggested by the numbers of unique molecules in the table). Still, it is encouraging to observe that the ration of DMSO unique moleculs versus 10uM of Etoposide is very similar in the two experiments. Suggesting that this is a robust feature that we are able to quantify, when we have enough material.


\section{Characterization of the errors in UMI}
Due to amplifications and sequencing, the original UMI molecule might 
be subject to errors and what we read in the fastq file is not what 
it is was supposed to be there at the beginning. It is important to implement 
an error-correction scheme in order to use more of the potentially 
useful reads. Error-correction has to be balanced with the ability to 
\textit{uniquely} recover reads and to avoid too many false-positives. 

To characterize the kind of errors which might generate faulty reads, we consider NC85 (approx 33M reads output from sequencing), which does not contain 
any UMI and it has a unique barcode which is supposed to be found before the 
cutsite. We select all the strings of 8bp which are found before a cutsite (allowing for 1 mismatch in the cutsite sequence), at the beginning of the fastq file. In order to keep the analysis simple we only consider mismatch errors, and do not consider insertion or deletion errors. Some of these strings found before the cutsite will be subject to mismatch errors with respect to the original barcode: 

\begin{tabular}{|c|c|}
\hline 
errors & total \\ 
\hline 
6149605 & 27563642 \\ 
\hline 
\end{tabular} 

The number of mismatches found in all the faulty barcodes is as follows:

\begin{tabular}{|c|c|}
\hline 
counts & number of MM \\ 
\hline 
4472524 & 1 \\ 
\hline 
1088701 & 2 \\ 
\hline 
410568 & 3 \\ 
\hline 
136272 & 4 \\ 
\hline 
34518 & 5 \\ 
\hline 
5972 & 6 \\ 
\hline 
847 & 7 \\ 
\hline 
203 & 8 \\ 
\hline 
\end{tabular}

as can be seen, grouping together faulty barcodes with 1 and 2 mismatches takes into account most of the mismatch-errors (90\% of the errors). This support our strategy to filter co-localized UMIs by identifying to the most frequent one those differing by at most 2bp.

%Basically, using as an error model the prefix sequence UMI-barcode[1,0,0]-cutsite[1,0,0]-genomic we can recover 78\% of the potentially useful reads. While using UMI-barcode[1,0,0]-cutsite[2,0,0]-genomic we might be able to recover 80\% of the potentially useful reads. While using UMI-barcode[2,0,0]-cutsite[2,0,0]-genomic we might be able to recover 84\% of the potentially useful reads.

A similar analysis can be performed using XZ9 (approx 2M reads from sequencing) where we have many copies of a reference BARCODE(16)-CUTSITE(6)-Genomic. This time we want to assess the effectiveness of the prefix filtering. In total there are 1900898 reads, of which 1274568 have at most 1 mismatch in the cutsite location and are preceded by 16bp. Of these 1274568, 1273545 have at most 1 mismatch in the 8bp 
preceding the cutsite. 
This analysis shows that 67\% of the reads show at most 1 mismatch in the cutsite location; while 66.9\% show at most 1 mismatch in the cutsite location and at most 1 mismatch in the barcode location. So, if we filter the initial fastq file for a prefix of the form UMI-barcode[1,0,0]-cutsite[1,0,0]-genomic we might loose approximately 30\% of the potentially useful reads. This percentage is not significantly lowered by allowing for more mismatches in the cutsite location or in the barcode location. 

%Of the latter 1273545, 1272537 have at most 1 mismatch 
%in the initial 8bp, and 1272876 have at most 2 mismatches in the initial 8bp. 

Probably one should take into account insertions and deletions in order to use more reads downstream in the pipeline. The problem with these extensions of the error model is that the downstream read identification is made more ambiguous. Hence, for the moment, we prefer to stick with a more simple to recover but robust to false-positives error model: UMI[2,0,0]-barcode[1,0,0]-cutsite[1,0,0]-genomic, where in the square bracket we quantify the allowed number of mismatches insertions and deletions, respectively.

\section{UMI scaling for RM82 and TK82}
In Fig.\ref{fig:umi_scaling_rm82_tk28} We consider the scaling of the number of locations having more than 
a given number of unique UMIs for RM82 and TK28.

\begin{figure}[hbtp]
\centering
\includegraphics[scale=1]{umi_scaling_rm82_tk28.pdf}
\caption{Log-log plot of the number of locations vs the minimum number of allowed UMIs.}
\label{fig:umi_scaling_rm82_tk28}
\end{figure}

While in the following table we show the 5-point statistics of the 
number of unique UMI found in a 5Kbp interval around different quartiles 
of the list of genes found in http://chromosome.sdsc.edu/mouse/download.html, for the expression profiles of the mouse liver. The data are obtained pulling 
of the RM82 and TK28 datasets together, and considering only locations with 
at least 2 UMI, in order to enhance the effect of biological DSB, assuming 
that for location having 1 UMI some of them will originate from natural DSB others from artificial DSB (while this is unlikely for locations having at least 2 UMI):
\\
\begin{tabular}{|c|c|c|c|c|c|c|}
\hline 
• & min & q1 & median & q3 & max & sum \\ 
\hline 
75-100 & 2 & 49 & 83 & 131 & 1112 & 392523 \\ 
\hline 
50-75 & 2 & 54 & 94 & 151 & 3686 & 448185 \\ 
\hline 
25-50 & 2 & 41 & 84 & 141 & 1370 & 393326 \\ 
\hline 
0-25 & 2 & 20 & 49 & 102 & 1164 & 267855 \\ 
\hline 
\end{tabular} 
\\
The following table is similar to the previous one, but it considers only 
locations having 1 UMI:
\\
\begin{tabular}{|c|c|c|c|c|c|c|}
\hline 
• & min & q1 & median & q3 & max & sum \\ 
\hline 
75-100 & 7 & 465 & 658 & 949 & 12759 & 3200318 \\ 
\hline 
50-75 & 2 & 471 & 687 & 1043 & 14890 & 3549092 \\ 
\hline 
25-50 & 6 & 405 & 625 & 976 & 8590 & 3149026 \\ 
\hline 
0-25 & 1 & 259 & 471 & 802 & 9088 & 2599510 \\ 
\hline 
\end{tabular} 
\\
The ratio between the medians in the top and bottom quartiles is 
higher for locations having at least 2 UMIs, consistently with the hypothesis 
that they better represent biological DSB instead of artificial ones. 

\section{Liver}
\subsection{DSB enrichment in highly expressed genes}
In this section we consider the following datasets: tk28-30-32 and rm82. Using the BED files containing the number of different UMI at each genome location, and the RNA expression data obtained from the Bing Ren lab (http://chromosome.sdsc.edu/mouse/download.html), we study the difference in 
DSB-location counts between different genes' groups. We first group genes according to 
different percentiles of their expression profile (top versus bottom 10\% expressed genes, or 20\%, 30\% and so on). For each group we evaluate the average number of DSBs found in a 1000bp window centered at TSS locations. We then calculate the ratio (of the top and bottom 10\% expressed genes for example) of these average values, and study their behavior as a function of 
the percentile parameter, for a fixed UMI lower threshold and experiment. 
\begin{figure}[hbtp]
\centering
\includegraphics[scale=1]{ratioVSperc_umi1_win500.pdf}
\caption{[Data in this plot have been obtained using the script TSS{\_}selective{\_}DSB{\_}enrichment.sh
located in the module directory of BLISS]
Ratio of the averaged DSB-location counts between highly and poorly expressed genes, for different percentiles. The UMI lower threshold is 1.}
\label{fig:topvsbot1}
\end{figure}
\begin{figure}[hbtp]
\centering
\includegraphics[scale=1]{ratioVSperc_umi2_win500.pdf}
\caption{[Data in this plot have been obtained using the script TSS{\_}selective{\_}DSB{\_}enrichment.sh
located in the module directory of BLISS]
Ratio of the averaged DSB-location counts between highly and poorly expressed genes, for different percentiles. The UMI lower threshold is 2.}
\label{fig:topvsbot2}
\end{figure}
One can argue that highly expressed genes are supposed to be more susceptible 
to DSBs (although it is not clear how much this depends on higher fragility rather then higher accessibility). The plots show that this effect is indeed quantitatively observed, and it is more evident for better preserved material. Moreover the genome locations having only 1 UMI are likely to be associated with random artefacts, rather than natural DSBs, as the increased values of the data shown in Fig.\ref{fig:topvsbot2} with respect to those in Fig.\ref{fig:topvsbot1} suggest. Another expected behavior, which we are able to capture quantitatively, is the monotonic decreasing of the curves with respect to the percentile parameter.

It is important to realize that we have considered the number of DSB locations and not the number of DSB per se. For these particular datasets (sequencing depth) this is a better choice because the estimated value for the averages has a lower error. If we considered the number of DSB instead, each location would bring an error in its estimate, and the average number of DSB per TSS would have an error scaling with the sum of the individual location errors. On the other hand, if we estimate the number of locations around a TSS the error would be much less (we need less reads to estimate locations than to estimate UMI per location). 

\subsection{Correlation between gene length and DSB counts}
We do not see any meaningful correlation, although this could be due to shallow sequencing. We count the number of DSB locations found in the neighbourhood (500bp) around TSS, or those found in the gene body. We bin (in units of 5Kbp) the lengths of the top 10\% expressed genes, and for each bin we plot the 5-point statistics. Results are shown in the following figures

\begin{figure}[hbtp]
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{dsbVSlen_body_top10perc_win500_binLen5k__tk30.pdf}
\caption{TK30. On the y axis we count the 5-point statistics for the number of DSB locations inside the genes belonging to bins of length equal to 5Kbp. The data has been obtained using DSBvsLENGHT{\_}gene.sh and the plot done using plot{\_}data.ipynb.}
\label{fig:dsbVSlen_body_top10perc_win500_binLen5k__tk30}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.5\textwidth}
\includegraphics[width=\linewidth]{dsbVSlen_body_top10perc_win500_binLen5k__tk32.pdf}
\caption{TK32. On the y axis we count the 5-point statistics for the number of DSB locations inside the genes belonging to bins of length equal to 5Kbp. There is a weak linear correlation with the gene length, which has to be expected because fragility is proportional to the region length. The data has been obtained using DSBvsLENGHT{\_}gene.sh and the plot done using plot{\_}data.ipynb.}
\label{fig:dsbVSlen_body_top10perc_win500_binLen5k__tk32}
\end{minipage}
\vspace*{0.5cm} % (or whatever vertical separation you prefer)
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{dsbVSlen_tss_top10perc_win500_binLen5k__tk30.pdf}
\caption{TK30. On the y axis we count the 5-point statistics for the number of DSB locations around the TSS (500bp) belonging to bins of length equal to 5Kbp. There is a stronger linear correlation with the gene length, which has to be expected because fragility is proportional to the region length. The data has been obtained using DSBvsLENGHT{\_}gene.sh and the plot done using plot{\_}data.ipynb.}
\label{fig:dsbVSlen_body_top10perc_win500_binLen5k__tk30}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{dsbVSlen_tss_top10perc_win500_binLen5k__tk32.pdf}
\caption{TK32. On the y axis we count the 5-point statistics for the number of DSB locations around the TSS (500bp) belonging to bins of length equal to 5Kbp. The data has been obtained using DSBvsLENGHT{\_}gene.sh and the plot done using plot{\_}data.ipynb.}
\label{fig:dsbVSlen_body_top10perc_win500_binLen5k__tk32}
\end{minipage}
\end{figure}

\subsection{Correlation between expression and DSB}
We do not see any meaningful correlation, although this could be due to shallow sequencing. We count the number of DSB locations found in the neighbourhood (500bp) around TSS, or those found in the gene body. We bin the expression (in unit of 100) of the top 10\% expressed genes, and for each bin we plot the 5-point statistics. Results are shown in the following figures

\begin{figure}[hbtp]
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{dsbVSexpr_body_top10perc_win500_binLen100__tk30.pdf}
\caption{TK30. On the y axis we count the 5-point statistics for the number of DSB locations inside the genes belonging to expression data binned on the x-axis. The data has been obtained using DSBvsEXPR{\_}gene.sh and the plot done using plot{\_}data.ipynb.}
\label{fig:dsbVSexpr_body_top10perc_win500_binLen100__tk30}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.5\textwidth}
\includegraphics[width=\linewidth]{dsbVSexpr_body_top10perc_win500_binLen100__tk32.pdf}
\caption{TK32. On the y axis we count the 5-point statistics for the number of DSB locations inside the genes belonging to expression data binned on the x-axis. The data has been obtained using DSBvsEXPR{\_}gene.sh and the plot done using plot{\_}data.ipynb.}
\label{fig:dsbVSexpr_body_top10perc_win500_binLen100__tk32}
\end{minipage}
\vspace*{0.5cm} % (or whatever vertical separation you prefer)
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{dsbVSexpr_tss_top10perc_win500_binLen100__tk30.pdf}
\caption{TK30. On the y axis we count the 5-point statistics for the number of DSB locations around the TSS (500bp) belonging to expression data binned on the x-axis. The data has been obtained using DSBvsLENGHT{\_}gene.sh and the plot done using plot{\_}data.ipynb.}
\label{fig:dsbVSexpr_body_top10perc_win500_binLen100__tk30}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{dsbVSexpr_tss_top10perc_win500_binLen100__tk32.pdf}
\caption{TK32. On the y axis we count the 5-point statistics for the number of DSB locations around the TSS (500bp) belonging to expression data binned on the x-axis. The data has been obtained using DSBvsLENGHT{\_}gene.sh and the plot done using plot{\_}data.ipynb.}
\label{fig:dsbVSexpr_body_top10perc_win500_binLen100__tk32}
\end{minipage}
\end{figure}

\subsection{DSB epigenomic enrichment}
[Data in this section are obtained using the script DSBvsFEAT.sh and  plot{\_}data.ipynb for plotting].
Given some epigenomic peak profile, we might wonder if there is any enrichment 
in DSB locations in the neighbourhood (1Kbp) of those peaks. Using liver data from the Bing Ren lab we can consider the following epigenomic signatures: 
enhancers, polII peaks (associated with polymerase activity), h3k4me3 peaks (associated with transcription-active promoters), h3k27ac peaks (associated with enhanced transcription activity) and ctcf peaks (associated with TADs boundaries). For each datasets we evaluate the 5-points statistics for the number of DSB locations found around the peaks. The following figures (from Fig.\ref{fig:enhancer} to Fig.\ref{fig:ctcf}) show the results (where the labels on the x-axis from 0 to 7 corresponds to the 
following experiments:  rm82-ACGACATC, rm82-CATCATCC, rm82-GTCGTTCC, rm82-TGATGATC, tk28-ACGACATC, tk28-GTCGTATC, tk30, tk32):
\begin{figure}[hbtp]
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{enhancer.pdf}
\caption{Enhancer. Raw counts. UMI $\geq 1$.}
\label{fig:enhancer}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.5\textwidth}
\includegraphics[width=\linewidth]{polII.pdf}
\caption{polII. Raw counts. UMI $\geq 1$.}
\label{fig:polII}
\end{minipage}
\vspace*{0.5cm} % (or whatever vertical separation you prefer)
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{h3k4me3.pdf}
\caption{h3k4me3. Raw counts. UMI $\geq 1$.}
\label{fig:h3k4me3}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{h3k27ac.pdf}
\caption{h3k27ac. Raw counts. UMI $\geq 1$.}
\label{fig:h3k27ac}
\end{minipage}
\vspace*{0.5cm} % (or whatever vertical separation you prefer)
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{ctcf.pdf}
\caption{ctcf. Raw counts. UMI $\geq 1$.}
\label{fig:ctcf}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{typical_rnd_boxPlot.pdf}
\caption{This is the typical box-plot originated by sampling at random 50K intervals, 20K times, along the genome, each of length 1Kbp and counting the number of DSB locations inside them. Each of the 20K samples will have a minimun, first quartile, median, and third quartile for the number of overlapping DSB locations. Taking the median of these 20K statistics we obtained the above box-plot (with 5-points statistics given by 0,0,1,2,225).}
\label{fig:typical_rnd_boxPlot}
\end{minipage}
\end{figure}

One could suspect that the values for these box-plots are somehow random and not very meaningful, indeed the opposite is true. A simple simulation (but quite time consuming so not yet available...) will show that placing peaks at random on the genome and counting the number of DSB locations in the surrounding region will produce a typical box-plot with min and first quartile equal to zeros, median equal to 1, third quartile equal to 2 and max value equal to 225 (these are the median values of 20K box-plots originated by placing 50K intervals, of 1Kbp, at random on the genome and counting the overlap with DSB locations). So a random box-plot (see Fig.\ref{fig:typical_rnd_boxPlot}) would be very different from the above epigenomic box-plots, meaning that there is correlation between epigenomic and fragile (and accessible) locations. 

\begin{figure}[hbtp]
\centering
\end{figure}


In order to compare different datasets and different epigenomic marks it is better to work with normalized data (with respect to the total number of DSB locations). So we repeat the previous analysis, this time considering the probability mass around each peak, and then evaluating the 5-point statistics. The obtained box-plots are shown in Fig.\ref{fig:enhancer_norm} to Fig.\ref{fig:ctcf_norm}, where we can appreciate the reproducibility in different datasets of the 5-point statistics for each epigenomic mark (discrepancies should be easily explained due to low count statistics), as well as the different enrichment for different epigenomic marks: polII and h3k4me3 seem to be the most enriched marks in DSB locations. 

\begin{figure}[hbtp]
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{enhancer_norm.pdf}
\caption{Enhancer. Probability plot. UMI $\geq 1$.}
\label{fig:enhancer_norm}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.5\textwidth}
\includegraphics[width=\linewidth]{polII_peak_norm.pdf}
\caption{polII. Probability plot. UMI $\geq 1$.}
\label{fig:polII_norm}
\end{minipage}
\vspace*{0.5cm} % (or whatever vertical separation you prefer)
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{h3k4me3_peak_norm.pdf}
\caption{h3k4me3. Probability plot. UMI $\geq 1$.}
\label{fig:h3k4me3_norm}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{h3k27ac_peak_norm.pdf}
\caption{h3k27ac. Probability plot. UMI $\geq 1$.}
\label{fig:h3k27ac_norm}
\end{minipage}
\vspace*{0.5cm} % (or whatever vertical separation you prefer)
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{ctcf_peak_norm.pdf}
\caption{ctcf. Probability plot. UMI $\geq 1$.}
\label{fig:ctcf_norm}
\end{minipage}
\end{figure}

Selecting only locations with at least 2 UMIs we have the statistics shown in Fig.\ref{fig:enhancer_norm_umi2} to Fig.\ref{fig:ctcf_norm_umi2}. In this case we have less counts, so only few datasets provide meaningful results, but the considerations for the case with UMI $\geq 1$ are still valid for UMI $\geq 2$.

\begin{figure}[hbtp]
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{enhancer_norm_umi2.pdf}
\caption{Enhancer. Probability plot. UMI $\geq 2$.}
\label{fig:enhancer_norm_umi2}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.5\textwidth}
\includegraphics[width=\linewidth]{polII_peak_norm_umi2.pdf}
\caption{polII. Probability plot. UMI $\geq 2$.}
\label{fig:polII_norm_umi2}
\end{minipage}
\vspace*{0.5cm} % (or whatever vertical separation you prefer)
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{h3k4me3_peak_norm_umi2.pdf}
\caption{h3k4me3. Probability plot. UMI $\geq 2$.}
\label{fig:h3k4me3_norm_umi2}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{h3k27ac_peak_norm_umi2.pdf}
\caption{h3k27ac. Probability plot. UMI $\geq 2$.}
\label{fig:h3k27ac_norm_umi2}
\end{minipage}
\vspace*{0.5cm} % (or whatever vertical separation you prefer)
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{ctcf_peak_norm_umi2.pdf}
\caption{ctcf. Probability plot. UMI $\geq 2$.}
\label{fig:ctcf_norm_umi2}
\end{minipage}
\end{figure}

\subsection{UMI filtering}
For a given dataset, we evaluate the ratio of the numbers of locations with at least 2 and at least 1 UMIs:
\begin{tabular}{|c|c|c|c|}
\hline 
\rule[-1ex]{0pt}{2.5ex} RM82 & TK28 & TK30 & TK32 \\ 
\hline 
\rule[-1ex]{0pt}{2.5ex} 0.0315 & 0.0253 & 0.0905 & 0.0360 \\ 
\hline 
\end{tabular} 

\subsection{TK30-31-32 expression-fragility relation}
We consider TK30-31-32, select the top and bottom 10{\%} expressed genes and for each we count the number of DSB locations found in a 5000bp-window centered around the corresponding TSS (Fig.\ref{fig:tssProfile_tk30_topVSbot}-\ref{fig:tssProfile_tk31_topVSbot}-\ref{fig:tssProfile_tk32_topVSbot}), and in the gene body (Fig.\ref{fig:mLIVER_tk30-body__boxplot_activeVSinactive}-\ref{fig:mLIVER_tk31-body__boxplot_activeVSinactive}-\ref{fig:mLIVER_tk32-body__boxplot_activeVSinactive}). 
\begin{figure}[hbtp]
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{tssProfile_tk30_topVSbot.pdf}
\caption{Profile of the breakome around TSS for active and inactive genes for TK30.}
\label{fig:tssProfile_tk30_topVSbot}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{tssProfile_tk31_topVSbot.pdf}
\caption{Profile of the breakome around TSS for active and inactive genes for TK31.}
\label{fig:tssProfile_tk31_topVSbot}
\end{minipage}
\vspace*{0.5cm} % (or whatever vertical separation you prefer)
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{tssProfile_tk32_topVSbot.pdf}
\caption{Profile of the breakome around TSS for active and inactive genes for TK32.}
\label{fig:tssProfile_tk32_topVSbot}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{mLIVER_tk30-body__boxplot_activeVSinactive.eps}
\caption{Boxplot comparison for active and inactive genes and their DSB count insider the gene-body for TK30.}
\label{fig:mLIVER_tk30-body__boxplot_activeVSinactive}
\end{minipage}
\vspace*{0.5cm}
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{mLIVER_tk31-body__boxplot_activeVSinactive.eps}
\caption{Boxplot comparison for active and inactive genes and their DSB count insider the gene-body for TK31.}
\label{fig:mLIVER_tk31-body__boxplot_activeVSinactive}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{mLIVER_tk32-body__boxplot_activeVSinactive.eps}
\caption{Boxplot comparison for active and inactive genes and their DSB count insider the gene-body for TK32.}
\label{fig:mLIVER_tk32-body__boxplot_activeVSinactive}
\end{minipage}

\end{figure}

\red{In order to study the relation between expression and fragility we consider the full list 
of genes' TSS (whose expression values are found from Bin Ren lab) for the mouse liver, and the number of breaks localized in a 1Kbp window around each TSS. We then bin the data into percentiles with respect to the expression values: 1st 10th percentile, 2nd 10th percentile, ...,10th 10th percentile. For each group we take the sum of the expression and the sum of the DSB in the group.  
Fig.\ref{fig:scatter_tk30-31-32_expression-fragility} clearly show a relation between {\bf E}xpression and {\bf F}ragility of the form $E \propto e^F$, supported as well by mESC data.}

\begin{figure}[hbtp]
 \centering
 \includegraphics[scale=1]{scatter_tk30-31-32_expression-fragility.pdf}
 \caption{On the y axis Log(FPKM), on the x axis the number of DSB in a 1Kbp window around the TSS (linear scale). The different curves come from TK30-31-32 dataset.}
 \label{fig:scatter_tk30-31-32_expression-fragility}
 \end{figure}
  

%Fig.\ref{fig:tk30_31_32_boxplot_DSBcounts_topANDbot_10perc} 
%show the result, where we can see the enrichment in the top 10\% of genes.

%\begin{figure}[hbtp]
%\centering
%\includegraphics[scale=0.5]{tk30_31_32_boxplot_DSBcounts_topANDbot_10perc.pdf}
%\caption{bla}
%\label{fig:tk30_31_32_boxplot_DSBcounts_topANDbot_10perc}
%\end{figure}

%In Fig.\ref{fig:tk30-31-32__topANDbot10perc__wind_dsbPerGene} we show the number of DSB locations per gene around 500bp regions whose distance from the TSS is given on the x-axis. 

%\begin{figure}[hbtp]
%\centering
%\includegraphics[scale=0.5]{tk30-31-32__topANDbot10perc__wind_dsbPerGene.pdf}
%\caption{Number of DSB per gene inside a 500bp window, for increasing distances from the TSS.}
%\label{fig:tk30-31-32__topANDbot10perc__wind_dsbPerGene}
%\end{figure}



\section{Mouse Embrionic Stem Cells}
We consider the dataset RM84 in BICRO26. We study the evolution of the breakome at 3 time-points for cells that start in an undifferentiated condition and then become neuronale differentiated cells. The time-points are 0-3-5 days of Retinoic Acid (RA) perturbation. 

The dataset we use to compare with expression profile comes from Bin Ren lab (mESC-zy27.gene.expr in the data subdirectory). After selecting the top and bottom 10 percent expressed genes, the number of DSB falling in the gene-body, and in the TSS 1Kbp window, are shown in Fig.\ref{fig:rm84-BODY_topVSbot10perc_undiff}-\ref{fig:rm84-TSS_topVSbot10perc_undiff}. There is a reproducible tendency to have more breaks in relation to highly expressed genes.

\begin{figure}[hbtp]
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{rm84-BODY_topVSbot10perc_undiff.pdf}
\caption{Comparison of number of DSB falling the gene-body (divided by gene-length), for top and bottom 10 percent genes and two replica, in undifferentiated cells.}
\label{fig:rm84-BODY_topVSbot10perc_undiff}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{rm84-TSS_topVSbot10perc_undiff.pdf}
\caption{Comparison of DSB falling the 1Kbp window around TSS, for top and bottom 10 percent genes, in undifferentiated cells.}
\label{fig:rm84-TSS_topVSbot10perc_undiff}
\end{minipage}
\end{figure}

To see how the breakome profile change during differentiation we can 
consider the entropy of the distribution of breaks around all TSS. Fig.\ref{fig:entropygap} shows how the difference between the maximum entropy ($Log(N_{tss})$, where $N_{tss}$ is the number of TSS 1Kbp neighbourhoods with at least 1 DSB ) and the actual entropy ($-\sum_{tss} p_{tss}Log p_{tss}$, where $p_{tss}$ is the probability of finding a DSB in that TSS window) -- which provides a measure of localization of the DSB distribution -- increases with time. This 
behavior supports a differentiation dynamics where specific genes (those identifying a cell) becomes more and more expressed, and as a consequence the DSB number around their TSS increases.

\begin{figure}[hbtp]
\centering
\includegraphics[scale=1]{entropygap.pdf}
\caption{Differentiation, as measured by the entropy gap (on the y-axis), increases with time (on the x-axis).}
\label{fig:entropygap}
\end{figure}

Fig.\ref{fig:tssprofile_RM84} shows the TSS profile around the top and bottom 10\% expressed genes for undifferentiated mESC. One can appreciate the high reproducibility of the profile, as well as the {recurrence in different dataset of peaks at few Kbp from the TSS origin: what are those peaks? Maybe they represent typical locations of enhancers which are fragile because it is required to break something before bringing them close to the promoter?} 

\begin{figure}[hbtp]
\centering
\includegraphics[scale=1]{tssprofile_RM84.pdf}
\caption{TSS profile for RM84 undifferentiated, with 2 replica. On the y axis we count the breaks, on the x axis we show the distance from the TSS. Note the peaks around 3Kbp from the TSS, are they fragile enhancers? What is the role of fragility around an enhancer, maybe plasticity in order to be brought close to the promoter.}
\label{fig:tssprofile_RM84}
\end{figure}

\red{In order to study the relation between expression and fragility we consider the full list 
of genes' TSS (whose expression values are found from Bin Ren lab) for the mouse ESC, and the number of breaks localized in a 1Kbp window around each TSS. We then bin the data into percentiles with respect to the expression values: 1st 10th percentile, 2nd 10th percentile, ...,10th 10th percentile. For each group we take the sum of the expression and the sum of the DSB in the group. Fig.\ref{fig:scatter_rm84_expressionVSfragility} clearly show a relation between {\bf E}xpression and {\bf F}ragility of the form $E \propto e^F$, supported as well by mouse liver data.}

\begin{figure}[hbtp]
 \centering
 \includegraphics[scale=1]{scatter_rm84_expressionVSfragility.pdf}
 \caption{On the y axis Log(FPKM), on the x axis the number of DSB in a 1Kbp window around the TSS (linear scale). The different curves come from the two replica for undifferentiated mESC.}
 \label{fig:scatter_rm84_expressionVSfragility}
 \end{figure}

\section{Human B-cells}
\subsection{Dataset: RM90/BICRO26} 
After 48h for the control sample and 48h for the stimulated sample we expect to see more DSBs after stimulation in the IGH locus S-regions, in particular in the $S\alpha1-2$ $S\gamma1-2-4$ and $S\mu$ sub-regions, while $S\epsilon$ should be depleted in DSB.

We create BED files with the locus and the S-regions coordinates. Using {\it bedtools intersect} we count the number of DSB locations -- in the control and in the treated samples -- found in the relevant feature-regions. Then we need to ask what is the significance of the treated-versus-control ratio that we observe (considering that due to apoptosis there will be a lot of random DSBs). To answer this question we generate 1000 random regions (using {\it bedtools shuffle}) on chr14 -- where the IGH locus is located -- of the same size as the relevant IGH feature-regions (excluding the centromeric and telomeric regions, as well as the relevant IGH feature-regions). For each of these random regions we count the number of DSB locations found in the control and in the treated sample. We then calculate their ratio and estimate its probability distribution (interpreting the ratio as a random variable). From this information we can estimate the significance of the treated-versus-control experimental ratio as the corresponding percentile in the probability distribution obtained from the Monte-Carlo simulation. For different feature-regions (entire IGH locus, the set of S-regions in the locus, the set of $S\alpha1-2$ $S\gamma1-2-4$ and $S\mu$ sub-regions, and finally the $S\epsilon$ region) the distribution always look like a Poisson distribution, which has to be expected since we are simply counting the number of random events occurring in a given interval. In Table \ref{tab:rm90_IGH_enrichment} it is shown the list of experimental values for the treated-versus-control ratio of DSB-location counts, and the corresponding percentiles in the distributions obtained from the Monte-Carlo simulations. The results in the table support the interpretation that the $S\alpha-S\gamma-S\mu$ sub-regions are indeed more prone to break than the $S\epsilon$ region, and also the fact that the entire LGH locus is more susceptible to break than similar regions (in size) on the rest of chromosome 14.

\begin{table}
\centering
\begin{tabular}{|c|c|c|}
\hline 
region & value & percentile \\ 
\hline 
IGH locus & 0.40 & 92th \\ 
\hline 
S-regions & 0.49 & 92th \\ 
\hline 
$S\alpha \cup S\gamma \cup S\mu$ & 0.53 & 94th \\ 
\hline 
\end{tabular} 
\caption{Experimental values for treated-versus-control DSB-location counts and corresponding percentile.}
\label{tab:rm90_IGH_enrichment}
\end{table}

\begin{table}
\centering
\begin{tabular}{|c|c|c|c|c|c|c|}
\hline 
region & value(C) & perc(C) & value(T) & perc(T) & T/C & perc(T/C) \\ 
\hline 
$S\mu$ & 20 & 75th & 7  & 70th & 0.35 & 47th\\
\hline
$S\gamma3$ & 43 & 96th & 20  & 98th & 0.46 & 70th\\
\hline
$S\gamma1$ & 25 & 97th & 19  & 99th & 0.76 & 83th\\
\hline
$S\alpha1$ & 7 & 45th & 2 & 45th & 0.28 & 35th\\
\hline
$S\gamma2$ & 23 & 98th & 13  & 99th & 0.56 & 76th\\
\hline
$S\gamma4$ & 43 & 98th & 23 & 99th & 0.53 & 76th\\
\hline
$S\epsilon$ & 35 & 81th & 12 & 75th & 0.34 & 45th\\
\hline
$S\alpha2$ & 8 & 63th & 5 & 75th & 0.62 & 80th\\
\hline
\end{tabular} 
\caption{Experimental values for treated and control samples (both absolute counts and percentiles), with DSB-location counts and treated-versus-control ratio, and corresponding percentiles.}
\label{tab:rm90_IGHregions_enrichment}
\end{table}

Another way to study the local effect of the treatment onto IGH S-regions is by subtracting the probability of finding a DSB location in the regions of the control from the probability in the treated sample. Figure \ref{fig:Sregions_probability_difference} shows the bar-plot for the S-regions. How meaningful are this numbers? Could they arise just by chance? To answer these questions we have to make some assumptions on the genome distribution of DSBs:
\begin{enumerate}
\item Random artificial DSB are distributed like a Poisson variable all over the genome, and given the presence of enough apoptotic cells they would be the vast majority.
\item Random artificial DSB are distributed in the same way for control and treated samples, and they would have mainly UMI=1 breaks.
\item Treatment-induced DSB should be localized in the loci where they are supposed to be found, and given enough sequencing depth they should have a stronger (not requiring too deep sequencing, compared to the following UMI-effect) DSB concentration effect, and a weaker (with respect to the previous effect, and detectable by higher depth sequencing) UMI-value effect, i.e. UMI$\geq 2$ should be more frequent than for random artificial DBSs.
\end{enumerate}
Taking this into account, and assuming we are not at saturation, we look for an increased probability of finding DSB locations in the regions where we are supposed to find more DSB after treatment: $Prob_T\left[loc\right]-Prob_C\left[loc\right]$. we quantify the significance of this metric as the significance of $Prob_T\left[loc\right]$, assuming that all the DSB in the 
control have a random and non-localized distribution (at least in the region of interest, and in any case this can be checked). The significance of $Prob_T\left[loc\right]$ is given by the corresponding percentile in the distribution of DSB-locations counts on random regions of similar size along the genome (only chromosome 14 in this case). Table \ref{tab:rm90_IGHregions_enrichment} and Fig.\ref{fig:Sregions_probability_difference} shows the 
raw data and their significance level (calculated as explained above). From these values it seems that we are not able to confidently study enrichment of too small regions, but we have to restrict the analysis to the ensemble of S regions or to the entire IGH locus, which indeed appear as enriched in DSB locations. 

\begin{figure}[hbtp]
\centering
\includegraphics[scale=1]{Sregions_probability_difference.pdf}
\caption{Difference between the DSB-location probabilities between treated and control. On the x-axis we plot $S\mu,S\gamma_3,S\gamma_1,S\alpha_1,S\gamma_2,S\gamma_4,S\epsilon,S\alpha_2$ from left to right.}
\label{fig:Sregions_probability_difference}
\end{figure}

\subsection{Dataset: RM96/BICRO27}
Here we repeat the analysis done previously for RM90, see the results in Table \ref{tab:rm96_IGHregions_enrichment} and in Table \ref{tab:rm96_larger_regions}

\begin{table}
\centering
\begin{tabular}{|c|c|c|c|c|c|c|}
\hline 
region & value(C) & perc(C) & value(T) & perc(T) & T/C & perc(T/C) \\ 
\hline 
$S\mu$ & 75 & 99th & 46  & 99th & 0.61 & 45th\\
\hline
$S\gamma3$ & 49 & 98th & 14  & 65th & 0.29 & 18th\\
\hline
$S\gamma1$ & 20 & 98th & 5  & 50th & 0.25 & 23th\\
\hline
$S\alpha1$ & 36 & 98th & 27 & 99th & 0.75 & 65th\\
\hline
$S\gamma2$ & 23 & 99th & 15  & 99th & 0.65 & 60th\\
\hline
$S\gamma4$ & 44 & 98th & 25 & 98th & 0.57 & 42th\\
\hline
$S\epsilon$ & 77 & 97th & 44 & 98th & 0.57 & 40th\\
\hline
$S\alpha2$ & 26 & 97th & 18 & 50th & 0.69 & 58th\\
\hline
\end{tabular} 
\caption{Experimental values for treated and control samples (both absolute counts and percentiles), with DSB-location counts and treated-versus-control ratio, and corresponding percentiles. RM96-BICRO27.}
\label{tab:rm96_IGHregions_enrichment}
\end{table}

\begin{table}
\centering
\begin{tabular}{|c|c|c|}
\hline 
region & value & percentile \\ 
\hline 
DOWN regulated genes & many & not-significant \\ 
\hline 
HOX genes &  & th \\ 
\hline 
Non-changing genes & many & not-significant \\ 
\hline 
UP regulated genes & many & not-significant \\ 
\hline 
$S\alpha \cup S\gamma \cup S\mu$ & 0.55 & 10th \\ 
\hline 
\end{tabular} 
\caption{Experimental values for treated/control DSB-location counts and corresponding percentile obtained from randomly shuffling regions of similar sizes.}
\label{tab:rm96_larger_regions}
\end{table}


\section{Fragility VS accessibility}
Fig.\ref{fig:fragilityVSaccessibility}-\ref{fig:fragilityVSaccessibility_rm84_rm94_undifferentiated}-\ref{fig:fragilityVSaccessibility_rm84_rm94_3days}-\ref{fig:fragilityVSaccessibility_rm84_rm94_5days}
compare the probability of finding a DSB and the probability of finding a restSeq-read in the same windows, after binning with a 100Kbp resolution. The figure shows that at fixed accessibility we have a wide range of values for the probability of finding a DSB, i.e. accessibility cannot explain fragility. Though of course accessibility helps in finding more DSB (see the bottom border of the point cloud). Moreover, the way in which cells are treated does not qualitatively modify this behaviour. It would be interesting to look into the 0 accessibility regions and see what characterize those DSB.
\begin{figure}[hbtp]
\centering
\begin{tabular}{@{}cc@{}}
  \includegraphics[width=.5\textwidth]{fragilityVSaccessibility.eps} &
  \includegraphics[width=.5\textwidth]{fragilityVSaccessibilitySuspension.eps}
\end{tabular}
\caption{Fragility in section nuclei (left) and suspension nuclei (right) versus accessibility}
\label{fig:fragilityVSaccessibility}
\end{figure}

%These figures has been obtained using /home/garner1/Work/pipelines/BLISS/bin/module/compare_TK32_RM92.sh,
%which produced: 
%/home/garner1/Work/pipelines/BLISS/EDA_summary/data/rm92-coverage-100000_.bed
%/home/garner1/Work/pipelines/BLISS/EDA_summary/data/tk32-coverage-100000_.bed
%with absolute molecule count per window 

\begin{figure}[hbtp]
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{fragilityVSaccessibility_rm84_rm94_undifferentiated.eps}
\caption{Fragility (y axis) versus accessibility (x axis), for all 100Kbp windows along the genome, and 2 replicated bliss experiments with the same restseq experiment. Undifferentiated mouse stem cells.}
\label{fig:fragilityVSaccessibility_rm84_rm94_undifferentiated}
\end{minipage}
\hspace{\fill}
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{fragilityVSaccessibility_rm84_rm94_3days.eps}
\caption{Fragility (y axis) versus accessibility (x axis), for all 100Kbp windows along the genome, and 2 replicated bliss experiments with the same restseq experiment. 3 days of differentiating mouse stem cells.}
\label{fig:fragilityVSaccessibility_rm84_rm94_3days}
\end{minipage}
\vspace*{0.5cm} % (or whatever vertical separation you prefer)
\begin{minipage}[t]{0.45\textwidth}
\includegraphics[width=\linewidth]{fragilityVSaccessibility_rm84_rm94_5days.eps}
\caption{Fragility (y axis) versus accessibility (x axis), for all 100Kbp windows along the genome, and 2 replicated bliss experiments with the same restseq experiment. 5 days of differentiating mouse stem cells.}
\label{fig:fragilityVSaccessibility_rm84_rm94_5days}
\end{minipage}
\end{figure}

\end{document}